packages/lib/file-api-driver-amazon-s3.js

```diff
@@ -3,6 +3,8 @@ const { basename } = require('./path-utils');
 const shim = require('./shim').default;
 const JoplinError = require('./JoplinError').default;
 const { Buffer } = require('buffer');
+const { Readable } = require('stream').Readable;
```

Comment:
The desktop app will load this fine.

on iOS I get `TypeError: undefined is not an object (evaluating '_$$_REQUIRE(_dependencyMap[8], "stream").Readable.Readable')` 

if I change it to `const Readable  = require('stream').Readable;` or `const { Readable }  = require('stream');`  I get undefined errors from the stream on iOS:

`[TypeError: stream.on is not a function. (In 'stream.on("data", function (chunk) {
            return chunks.push(chunk);
          })', 'stream.on' is undefined)]`

What am I missing?

Reply:
seems this could fix it: https://github.com/laurent22/joplin/pull/5212#issuecomment-883691212

Reply:
I've updated my comment above with a SO link

With that the app starts but fails to connect returning `SignatureDoesNotMatch`. I did put invalid password initially but I think I have fixed it since, and yet..
I'm going to reinstall the app and see if it helps.

Btw, @laurent Joplin probably shouldn't do this (or maybe it's caused by the PR):
```
[Tue Jul 20 2021 21:44:47.477]  INFO     Saving sync.8.password <plain text password here>
```

Reply:
afaik, the password logging bit happened before this change as I saw it in early troubleshooting.

Reply:
Indeed, it's not related to your change.
It's here: https://github.com/laurent22/joplin/blob/366ef025dca9aad7c4254f198d5ef0d85e5b5bd4/packages/lib/components/shared/config-shared.js#L88

Reply:
Thanks for spotting this, it's debugging code that should have been removed. Actually could you remove it in this pr while you're at it?

Reply:
There's `SettingItem.secure` that can potentially be used to mask/filter log. Though I'm not sure if that value is available at the place where it's logged.

Reply:
Solved in my latest commit.